From 5a674d684d58d89fc0e4bdfe1c1a5044f85a1089 Mon Sep 17 00:00:00 2001
From: Bastian Bittorf <bittorf@bluebottle.com>
Date: Sun, 17 Feb 2013 16:16:21 +0100
Subject: [PATCH] [packages] olsrd: don't try to start smartgw if not all deps are installed

[packages] olsrd: don't try to start smartgw if not all deps are installed
if kmod-ipip is missing, there is no need to invoke an expensive operation.

Signed-off-by: Bastian Bittorf <bittorf@bluebottle.com>
---
 net/olsrd/files/olsrd.init |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/net/olsrd/files/olsrd.init b/net/olsrd/files/olsrd.init
index fbdda30..0656247 100644
--- a/net/olsrd/files/olsrd.init
+++ b/net/olsrd/files/olsrd.init
@@ -563,6 +563,7 @@ olsrd_setup_smartgw_rules() {
 	# Check if ipip is installed
 	[ -e /etc/modules.d/[0-9]*-ipip ] || {
 		log "$funcname() Warning: kmod-ipip is missing. SmartGateway will not work until you install it."
+		return 1
 	}
 
 	wanifnames=$(ip r l e 0/0 t all | sed -e 's/^.* dev //' |cut -d " " -f 1 | sort | uniq)
@@ -713,6 +714,7 @@ start() {
 			service_start /usr/sbin/olsrd -f "$OLSRD_CONFIG_FILE" -nofork
 		fi
 	fi
+
 	olsrd_setup_smartgw_rules
 }
 
-- 
1.7.2.5

